src/lib.rs

```diff
@@ -95,27 +102,44 @@ enum ConfigAction {
     EXIT,
 }
 
+impl std::fmt::Display for ConfigAction {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        std::fmt::Debug::fmt(self, f)
+    }
+}
+
 pub struct ConfigEvent {
     action: ConfigAction,
     config_ids: Vec<String>,
     config_str: String,
     stage_id: u32,
 }
 
+pub fn setup_logging() {
+    let timer = tracing_subscriber::fmt::time::time();
+    let collector = tracing_subscriber::fmt()
+        .with_max_level(Level::DEBUG)
+        // disable color to make CLion happy
+        .with_ansi(false)
+        .with_thread_ids(true)
+        .with_timer(timer)
+        .finish();
+    tracing::subscriber::set_global_default(collector).expect("setting default subscriber failed");
```

Comment:
What's the impact of `set_global_default` here? I see there are same usage in vector's code for something like metrics collection. I'm not sure if this will block vector's original functional logic.
As I see in this method's doc(https://docs.rs/tracing/0.1.37/tracing/subscriber/fn.set_global_default.html):
```Can only be set once; subsequent attempts to set the global default will fail. Returns whether the initialization was successful.```. We should make sure not break something else.

Reply:
https://github.com/vectordotdev/vector/search?q=set_global_default it is only used in test/benches of vector, so it should be fine